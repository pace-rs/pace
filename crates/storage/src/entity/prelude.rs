//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.15

pub use super::activities::Entity as Activities;
pub use super::activities_categories::Entity as ActivitiesCategories;
pub use super::activities_tags::Entity as ActivitiesTags;
pub use super::activity_kinds::Entity as ActivityKinds;
pub use super::activity_status::Entity as ActivityStatus;
pub use super::categories::Entity as Categories;
pub use super::descriptions::Entity as Descriptions;
pub use super::tags::Entity as Tags;
