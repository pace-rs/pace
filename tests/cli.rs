use assert_cmd::Command;
// use pace_core::ActivityLog;
use predicates::prelude::predicate;
// use tempfile::tempdir;

pub type TestResult<T> = Result<T, Box<dyn std::error::Error>>;

pub fn pace_runner(/*temp_dir: &TempDir*/) -> TestResult<Command> {
    // TODO: when we have implemented init, we can use this to create a new pace project
    // let _repo_dir = temp_dir.path();

    let runner = Command::new(env!("CARGO_BIN_EXE_pace"));

    Ok(runner)
}

// TODO: when we have implemented init, we can use this to create a new pace project
// fn setup() -> TestResult<TempDir> {
//     let temp_dir = tempdir()?;
//     pace_runner(&temp_dir)?.args(["init"]).assert().success();

//     Ok(temp_dir)
// }

#[test]
fn test_version_command_passes() -> TestResult<()> {
    _ = pace_runner()?
        .arg("--version")
        .assert()
        .success()
        .stdout(predicate::str::contains(env!("CARGO_PKG_VERSION")));

    Ok(())
}

#[test]
fn test_help_command_passes() -> TestResult<()> {
    _ = pace_runner()?
        .arg("--help")
        .assert()
        .success()
        .stdout(predicate::str::contains("Usage:"));

    Ok(())
}

// TODO!: Test begin command
// #[test]
// fn test_begin_command_passes() -> TestResult<()> {
//     let activity_log_file = tempdir()?.into_path().join("activity_log.toml");

//     let desc = "Test description";
//     let category = "Test::Category";
//     let time = "22:00";

//     if activity_log_file.exists() {
//         std::fs::remove_file(&activity_log_file)?;
//     }

//     std::fs::write(&activity_log_file, "")?;

//     _ = pace_runner()?
//         .args([
//             "-a",
//             activity_log_file.as_path().to_str().unwrap(),
//             "begin",
//             desc,
//             "-c",
//             category,
//             "-t",
//             time,
//         ])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("started"));

//     let contents = std::fs::read_to_string(&activity_log_file)?;

//     let activity_log = toml::from_str::<ActivityLog>(&contents)?;

//     assert_eq!(activity_log.activities().len(), 1);

//     let activity = activity_log.activities().front().unwrap();

//     assert_eq!(activity.description(), &Some(desc.to_string()));
//     assert_eq!(activity.category(), &Some(category.to_string()));
//     assert_eq!(format!("{:?}", activity.begin()), time);

//     if activity_log_file.exists() {
//         std::fs::remove_file(&activity_log_file)?;
//     }

//     Ok(())
// }

// TODO: Test end command
// #[test]
// fn test_end_command_passes() -> TestResult<()> {
//     pace_runner()?
//         .args(["-a", "./activity_log.toml", "end"])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("finished")); // TODO

//     Ok(())
// }

// TODO: Test now command
// #[test]
// fn test_now_command_passes() -> TestResult<()> {
//     pace_runner()?
//         .args(["-a", "./activity_log.toml", "now"])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("current")); // TODO

//     Ok(())
// }
