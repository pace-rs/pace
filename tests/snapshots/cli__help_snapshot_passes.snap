---
source: tests/cli.rs
assertion_line: 66
info:
  program: pace
  args:
    - help
---
success: true
exit_code: 0
----- stdout -----
pace - Mindful Time Tracking: Simplify Your Focus and Boost Productivity Effortlessly.

Usage: pace [OPTIONS] <COMMAND>

Commands:
  adjust  üìù Adjust the details of an activity, such as its category, description, or tags
              [aliases: a]
  begin   ‚åö Starts tracking time for an activity [aliases: b]
  end     ‚èπÔ∏è  Stops time tracking for the most recent or all activities [aliases: e]
  hold    ‚è∏Ô∏è  Pauses the time tracking for the most recent active activity [aliases: h]
  now     ‚è≤Ô∏è  Shows you at a glance what you're currently tracking [aliases: n]
  resume  ‚èØÔ∏è  Resumes a previously paused activity, allowing you to continue where you left off
              [aliases: r]
  review  üìà Get sophisticated insights on your activities [aliases: rev]
  setup   üõ†Ô∏è  Set up a pace configuration, a new project, or generate shell completions
              [aliases: s]
  docs    üìö Open the online documentation for pace [aliases: d]
  help    Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose
          Enable verbose logging
      --config <CONFIG>
          Use the specified config file [env: PACE_CONFIG_FILE=]
      --activity-log-file <ACTIVITY_LOG_FILE>
          Use the specified activity log file [env: PACE_ACTIVITY_LOG_FILE=]
      --home <HOME>
          Pace Home Directory [env: PACE_HOME=]
  -h, --help
          Print help
  -V, --version
          Print version

----- stderr -----
